Fix inspector showing Projection fields in wrong order

MP430 opened the pull request:
Hi,

When the inspector displays a projection, either as an `@export` from a script or a shader uniform, the columns and rows are the wrong way around/transposed.

I've fixed this by just swapping the row numbering in inspector_properties.cpp. The ordering is now consistent with i.e. Basis.


Here's the scene I used to test it also:
```
[gd_scene load_steps=5 format=3 uid="uid://dedlr02a8xi0v"]

[sub_resource type="BoxMesh" id="BoxMesh_4p3ci"]

[sub_resource type="Shader" id="Shader_31pbg"]
code = "shader_type spatial;

uniform mat4 testMatrix;

void vertex() {
	vec4 result = testMatrix * vec4(VERTEX.xyz,1);
	VERTEX = result.xyz / result.w;
}
"

[sub_resource type="ShaderMaterial" id="ShaderMaterial_vyl1g"]
render_priority = 0
shader = SubResource("Shader_31pbg")
shader_parameter/testMatrix = Projection(1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1)

[sub_resource type="GDScript" id="GDScript_4p3ci"]
script/source = "extends MeshInstance3D

@export var inVec: Vector4
@export var inProj: Projection
@export var outProj: Projection

func _ready() -> void:
	outProj = inProj

func _process(delta: float) -> void:
	print(inProj * inVec)
"

[node name="Node3D" type="Node3D"]

[node name="MeshInstance3D" type="MeshInstance3D" parent="."]
mesh = SubResource("BoxMesh_4p3ci")
surface_material_override/0 = SubResource("ShaderMaterial_vyl1g")
script = SubResource("GDScript_4p3ci")
inVec = Vector4(1, 1, 1, 1)

```

I love Godot by the way!

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

MP430 committed:
Fix inspector showing Projection fields in wrong order

When the inspector displays a projection, either as an @export from a script or a shader uniform, the columns and rows are the wrong way around/transposed.

I've fixed this by just swapping the row numbering in inspector_properties.cpp.
The ordering is now consistent with i.e. Basis.

AThousandShips commented:
This change is not correct, this swaps the labels, which match those of the class, before they went `xx`, `xy`, ..., `wz`, `ww`, matching those of the class, now it goes `xx`, `yx`, ..., `zw`, `ww` which breaks the relationship

AThousandShips reviewed (dismissed):
See above, the ordering of the array on line 2703 needs to be adjusted to match

MP430 commented:
Hmm, I'm not sure about that.

With Basis, the xy label corresponds to Basis.y.x. Example:
<img width="1194" height="379" alt="image" src="https://github.com/user-attachments/assets/3e946b07-6813-40cd-a8b0-33933f1d4645" />

This seems intended, because the documentation for Basis says that they're exposed in column-major order.

The documentation also says that Projections are column-major. But, this was backwards.

Before my change, I see this, which is unexpected because it's different to Basis:
<img width="1196" height="374" alt="image" src="https://github.com/user-attachments/assets/60bc6ded-5eac-4a24-91af-732b67c6dabb" />


After my change, I see this, which is consistent with Basis:
<img width="1193" height="397" alt="image" src="https://github.com/user-attachments/assets/cb286042-2e5e-41e3-b464-5b685775ff75" />

So I think this change is correct. Unless the labels for the Basis are also incorrect?

MP430 commented:
I think projection is definitely wrong, unfortunately, I don't know about the labelling though, maybe that should be swapped everywhere.

Another example, using an identity matrix, but with a 1 in the top-right. I'd expect this to create a translation. 
Before change, I get a projection instead:
<img width="1087" height="794" alt="image" src="https://github.com/user-attachments/assets/57f50a49-d212-4f85-8a96-63932b01b5cb" />

After change, I get a translation as expected:
<img width="1067" height="737" alt="image" src="https://github.com/user-attachments/assets/7329b5a8-9db1-4232-bd96-cb14c2e25a6a" />

So at least, the positions are wrong. I don't know whether the labelling should be different though.

AThousandShips commented:
Accessing `Projection` and `Basis` in the same way is not matching the data though? `basis[2][1]` is the same as `projection.columns[1][2]` though, so swapping these makes the data not represent the data correctly

Try using the string representation, it should print:
`[X: (0, 2, 0, 0), Y: (1, 0, 0, 0), Z: (0, 0, 0, 0), W: (0, 0, 0, 0)]`
And
`[X: (0, 2, 0), Y: (1, 0, 0), Z: (0, 0, 0)]`
If your changes are correct

AThousandShips commented:
But that depends on how it's meant to be represented, I'd suggest checking the actual data to see that both actually print the same when processed directly

MP430 commented:
I don't have a strong opinion on that, but I would be happy to swap the labels xy->yx on all the types, if you think that's best?

MP430 committed:
Swapped xy-yx etc. labels on Transform2D, Transform3D, Basis, Projection in inspector

The labels now match the column-major access as provided to gdscript.
I.E. Basis.x.y corresponds to the xy label in the inspector.

MP430 commented:
Hi, 

I've updated the PR with new labels for Transform2D, Transform3D, Basis, and Projection. 
They're now all consistent and match how they're exposed in gdscript. 
Here's an example showing how they all display now:
<img width="1325" height="788" alt="image" src="https://github.com/user-attachments/assets/12cc2fe2-06e0-433d-8f51-14e2d97b44a5" />

Hopefully that was the right thing to do, but let me know if not.
Thanks!
